semantic/semantic.fbs

```diff
@@ -0,0 +1,354 @@
+    init_:Expression; 
+}
+
+table MemberAssignment {
+    loc:SourceLocation; 
+    member:MemberExpression;
+    init_:Expression; 
+}
+
+table Extern {
+    loc:SourceLocation; 
+    assignments:[ExternalVariableAssignment];
+    block:ExternBlock; 
+}
+
+table ExternalVariableAssignment {
+    loc:SourceLocation; 
+    identifier:Identifier;
+}
+
+table ExternBlock {
+    loc:SourceLocation;
+    node:Block;
+}
+
+// Expressions
+
+// Unions cannot be vector elements,
+// so wrap the union in a table.
+table WrappedExpression {
+    expression:Expression;
+}
+
+table StringExpression {
+    loc:SourceLocation;
+    parts:[StringExpressionPart];
+}
+
+table StringExpressionPart {
+    loc:SourceLocation;
+    // Following two fields are mutually exclusive:
+    text_value:string; // for TextPart
+    interpolated_expression:Expression; // for InterpolatedPart
+}
+
+table ArrayExpression {
+    loc:SourceLocation;
+    elements:[WrappedExpression];
+}
+
+table FunctionExpression {
+    loc:SourceLocation;
+    defaults:ObjectExpression;
+    Block:FunctionBlock;
+}
+
+table FunctionBlock {
+    loc:SourceLocation;
+    parameters:FunctionParameters;
+    body:Block;
+}
+
+table FunctionParameters {
+    loc:SourceLocation;
+    list:[FunctionParameter];
+    pipe:Identifier;
+}
+
+table FunctionParameter {
+    loc:SourceLocation;
+    key:Identifier;
+}
+
+enum Operator : byte {
+  MultiplicationOperator,
+  DivisionOperator,
+  ModuloOperator,
+  PowerOperator,
+  AdditionOperator,
+  SubtractionOperator,
+  LessThanEqualOperator,
+  LessThanOperator,
+  GreaterThanEqualOperator,
+  GreaterThanOperator,
+  StartsWithOperator,
+  InOperator,
+  NotOperator,
+  ExistsOperator,
+  NotEmptyOperator,
+  EmptyOperator,
+  EqualOperator,
+  NotEqualOperator,
+  RegexpMatchOperator,
+  NotRegexpMatchOperator,
+  InvalidOperator,
+}
+
+enum LogicalOperator : byte {
+  AndOperator,
+  OrOperator,
+}
+
+table BinaryExpression {
+    loc:SourceLocation;
+    operator:Operator; 
+    left: Expression;
+    right: Expression;
+}
+
+table CallExpression {
+    loc:SourceLocation;
+    callee:Expression;
+    arguments:[ObjectExpression];
+    pipe:Expression;
+}
+
+table ConditionalExpression {
+    loc:SourceLocation;
+    test:Expression;
+    alternate:Expression;
+    consequent:Expression;
+}
+
+table LogicalExpression {
+    loc:SourceLocation;
+    operator:LogicalOperator;
+    left:Expression;
+    right:Expression;
+}
+
+table MemberExpression {
+    loc:SourceLocation;
+    object:Expression;
+    property:string;
+}
+
+table IndexExpression {
+    loc:SourceLocation;
+    array:Expression;
+    index:Expression;
+}
+
+table ObjectExpression {
+    loc:SourceLocation;
+    with:IdentifierExpression;
+    properties:[Property];
+}
+
+table UnaryExpression {
+    loc:SourceLocation;
+    operator:Operator;
+    argument:WrappedExpression;
+}
+
+table Property {
+    loc:SourceLocation;
+    key:PropertyKey; 
+    value:WrappedExpression;
```

Comment:
Same here, I don't think this needs to be wrapped.  I think unions only need to be wrapped if they are in an array.